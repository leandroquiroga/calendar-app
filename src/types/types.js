// Objeto que mantiene centralizado todos los tipos de la acciones


export const types = {
  uiOpenModal: '[ui] Open modal',
  uiCloseModal: '[ui] Close modal',

  eventAddNew: '[event] Add new',
  eventSetActive: '[event] Set active',
  eventUpdateNotes: '[event] Updated notes',
  clearActiveEvent: '[event] Clear notes',
  eventDeleteNotes: '[event] Delete notes'
}
