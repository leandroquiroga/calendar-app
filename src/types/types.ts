export interface TypesRedux {
  uiOpenModal: string ;
  uiCloseModal: string ;
  eventStartAddNew: string ;
  eventAddNew: string ;
  eventSetActive: string ;
  eventUpdateNotes: string ;
  clearActiveEvent: string ;
  eventDeleteNotes: string ;
  eventLoaded: string ;
  eventLogout: string ;
  authChekingFinish: string ;
  authStartLogin: string ;
  authLogin: string ;
  authStartRegister: string ;
  authStartTokenRenew: string ;
  authLogout: string ;
}


// Objeto que mantiene centralizado todos los tipos de la acciones
export const types: TypesRedux = {
  uiOpenModal: "[ui] Open modal",
  uiCloseModal: "[ui] Close modal",

  eventStartAddNew: "[event] Start add new",
  eventAddNew: "[event] Add new",
  eventSetActive: "[event] Set active",
  eventUpdateNotes: "[event] Updated notes",
  clearActiveEvent: "[event] Clear notes",
  eventDeleteNotes: "[event] Delete notes",
  eventLoaded: "[event] Loaded notes",
  eventLogout: "[event] Logout notes",

  authChekingFinish: "[auth] Finish checking state",
  authStartLogin: "[auth] Start login",
  authLogin: "[auth] Login",
  authStartRegister: "[auth] Start register",
  authStartTokenRenew: "[auth] Start token renew",
  authLogout: "[auth] Logout",
};

export default types;