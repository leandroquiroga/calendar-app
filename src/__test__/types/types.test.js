import { types } from "../../types/types";

describe('Test in Types.js', () => {
  test('This test must be equal a object', () => {
    expect(types).toEqual({
      uiOpenModal: '[ui] Open modal',
      uiCloseModal: '[ui] Close modal',
    
      eventStartAddNew: '[event] Start add new',
      eventAddNew: '[event] Add new',
      eventSetActive: '[event] Set active',
      eventUpdateNotes: '[event] Updated notes',
      clearActiveEvent: '[event] Clear notes',
      eventDeleteNotes: '[event] Delete notes',
      eventLoaded: '[event] Loaded notes',
      eventLogout: '[event] Logout notes',
    
      authChekingFinish: '[auth] Finish checking state',
      authStartLogin: '[auth] Start login',
      authLogin: '[auth] Login',
      authStartRegister: '[auth] Start register',
      authStartTokenRenew: '[auth] Start token renew',
      authLogout: '[auth] Logout',
    });
  });

  test('This test must retunt a capture of the object', () => {
    expect(types).toMatchSnapshot()
  });
});