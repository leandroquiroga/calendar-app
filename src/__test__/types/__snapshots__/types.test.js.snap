// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Types.js This test must retunt a capture of the object 1`] = `
Object {
  "authChekingFinish": "[auth] Finish checking state",
  "authLogin": "[auth] Login",
  "authLogout": "[auth] Logout",
  "authStartLogin": "[auth] Start login",
  "authStartRegister": "[auth] Start register",
  "authStartTokenRenew": "[auth] Start token renew",
  "clearActiveEvent": "[event] Clear notes",
  "eventAddNew": "[event] Add new",
  "eventDeleteNotes": "[event] Delete notes",
  "eventLoaded": "[event] Loaded notes",
  "eventLogout": "[event] Logout notes",
  "eventSetActive": "[event] Set active",
  "eventStartAddNew": "[event] Start add new",
  "eventUpdateNotes": "[event] Updated notes",
  "uiCloseModal": "[ui] Close modal",
  "uiOpenModal": "[ui] Open modal",
}
`;
